.github/workflows/deploy-to-k8s.yaml

```diff
@@ -0,0 +1,68 @@
+name: deploy-to-k8s
+on:
+  workflow_call:
+    inputs:
+      KUBECTL_VERSION:
+        description: 'kubectl version'
+        type: string
+        required: true
+      NAMESPACE:
+        description: 'namespace of the k8s cluster'
+        type: string
+        required: true
+      SECRET_FROM_ENV_PATH:
+        description: 'path to the k8s secret from env yaml file'
+        type: string
+      SECRET_NAME:
+        description: 'name for the k8s secret'
+        type: string
+      MANIFEST_OR_FOLDER_PATH:
+        description: 'path to the manifest yaml file or folder'
+        type: string
+        required: true
+    secrets:
+      KUBE_CONFIG_BASE64:
+        description: 'kube config file in base64 encoded'
+        required: true
+
+jobs:
+  deploy-k8s:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout caller repo
+        uses: actions/checkout@v4
+      - name: Setup kubectl
+        uses: azure/setup-kubectl@v3
+        with:
+          version: ${{ inputs.KUBECTL_VERSION }}
+        id: install
+      - name: Get KUBECONFIG
+        id: get_kubeconfig
+        run: |
+          echo "${{ secrets.KUBE_CONFIG_BASE64 }}" | base64 -d >> ${{ github.workspace }}/kubeconfig
+          KUBECONFIG=${{ github.workspace }}/kubeconfig
+          echo "::add-mask::$KUBECONFIG"
+          echo "KUBECONFIG=$KUBECONFIG" >> $GITHUB_OUTPUT
+      - name: Check existing resources in the namespace
+        env:
+          KUBECONFIG: ${{steps.get_kubeconfig.outputs.KUBECONFIG}}
+        run: |
+          # create namespace if not exists
+          if [ -z "$(kubectl get ns | grep ${{ inputs.NAMESPACE }})" ]; then
+            kubectl create ns ${{ inputs.NAMESPACE }}
+            echo "INFO:: Created namespace ${{ inputs.NAMESPACE }}"
+          fi
+          kubectl -n ${{ inputs.NAMESPACE }} get all
+      - name: Time window to manually cancel before deployment
+        run: |
+          sleep 30
```

Comment:
Consider replacing with a "pause for manual approval" step: https://github.com/marketplace/actions/manual-workflow-approval

Reply:
Oh, glad to know! I thought this is an enterprise feature

Reply:
@ethho This action hacks this by opening a Github Issue, but by the time this issue has been created till it gets approved or rejected, Github Actions workflow are staying at running state. This is ok for public repo, but it's going to waste a lot of Github Actions running quota and after it ran out, we have to pay for those minutes.
[This](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments) is the one I mentioned, but it requires 'GitHub Pro, GitHub Team, or GitHub Enterprise' to use in private repos.

Reply:
Actually, I just realized, we recently upgraded datajoint-company to Github Team! Let me try that feature!

Reply:
Nvm... This review feature only available in Github Enterprise now for private repos, [here](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#required-reviewers): '_Note: If you are on a GitHub Free, GitHub Pro, or GitHub Team plan, required reviewers are only available for public repositories._'

Reply:
> This is ok for public repo(since the runtime is free), but in private repo, it's going to waste a lot of Github Actions running quota and after it ran out, we have to pay for those minutes. 

This is a good point. To address this problem, we could limit the Actions minutes used by setting a [`timeout-minutes`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes) on the step. Of course, this forces a review within `x` minutes, which is a bit inconvenient for the reviewers.

From https://github.com/marketplace/actions/manual-workflow-approval:

> This is a very common feature for a deployment or release pipeline, and while this functionality is available from GitHub, it requires the use of environments and if you want to use this for private repositories then you need GitHub Enterprise. This action provides manual approval without the use of environments, and is freely available to use on private repositories.

As I understand it, this workflow provides a workaround that _avoids_ using [the Enterprise feature: environments for deployments (and all the features that are included such as required reviewers)](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments) that you mentioned. I haven't tried this workflow, but as advertised, it should provide the same functionality as the Enterprise feature, but free for all private repos. 

----

Overall, I think you've made some good points. I'm realizing that it's going to take more effort to set up this additional workflow, and since this is low priority, we can leave it with the `sleep` for now.